platform/ios/src/MLNMapView.h

```diff
@@ -1253,6 +1253,25 @@ MLN_EXPORT
  */
 - (void)flyToCamera:(MLNMapCamera *)camera withDuration:(NSTimeInterval)duration peakAltitude:(CLLocationDistance)peakAltitude completionHandler:(nullable void (^)(void))completion;
 
+
+/**
+ Moves the viewpoint to a different location using a transition animation that
+ evokes powered flight and an optional transition duration and peak altitude.
+
+ The transition animation seamlessly incorporates zooming and panning to help
+ the user find his or her bearings even after traversing a great distance.
+
+ @param camera The new viewpoint.
+ @param duration The amount of time, measured in seconds, that the transition
+    animation should take. Specify `0` to jump to the new viewpoint
+    instantaneously. Specify a negative value to use the default duration, which
+    is based on the length of the flight path.
+ @param edgePadding The minimum padding (in screen points) that would be visible
+ around the returned camera object if it were set as the receiver’s camera.
+ @param completion The block to execute after the animation finishes.
+ */
+- (void)flyToCamera:(MLNMapCamera *)camera edgePadding:(UIEdgeInsets)insets withDuration:(NSTimeInterval)duration completionHandler:(nullable void (^)(void))completion;
```

Comment:
> @1ec5 Do you know why something like this was never added?

I think by the time we added the flight methods, we were already reeling from how many different versions of `-setCamera:…` we needed to support. (Oh if only we could have written this library in pure Swift to begin with!) We tried to keep the number of variations of `-flyToCamera:…` to a minimum. At the time, we had a practical use case for publicly setting edge padding with `-setCamera:…`, for keeping something anchored on screen while zooming, but not for publicly setting edge padding with `-flyToCamera:…`. But internally, we did call essentially this method when jumping to the user’s location from afar:

https://github.com/maplibre/maplibre-native/blob/0af3f21367a4cb044fdd55fcb629dc7ee29fa765/platform/ios/src/MLNMapView.mm#L6229-L6233

It wasn’t until later that developers started asking for this particular method: https://github.com/mapbox/mapbox-gl-native/issues/8625#issuecomment-451557016.

Reply:
Yeah, swift would be nice. I created a discussion for that if you are interested: https://github.com/maplibre/maplibre-native/discussions/1460

Agree, there are too many different methods to move a camera to position X with parameters Y.
This is also a problem of the C library and I have the feeling that the native android and iOS sdks just expose whatever people needed. This is no expection because we also just exposed this method because we need it.

The question is how to avoid that? Having a smaller API surface? But how can we make sure that all use cases are covered? 🤔